src/main/scala/com/evolutiongaming/retry/OnError.scala

```diff
@@ -25,13 +25,22 @@ object OnError {
       }
   }
 
+  def fullLogOnRetry[F[_]](log: Log[F]): OnError[F, Throwable] =
+    onRetry(log) { (error, delay) =>
+      log.warn(s"failed, retrying in $delay, error: $error", error)
+    }
 
-  def fromLog[F[_]](log: Log[F]): OnError[F, Throwable] = {
-    (error: Throwable, status: Retry.Status, decision: Decision) => {
 
+  def fromLog[F[_]](log: Log[F]): OnError[F, Throwable] =
+    onRetry(log) { (error, delay) =>
+      log.warn(s"failed, retrying in $delay, error: $error")
+    }
+
+  private def onRetry[F[_]](log: Log[F])(callback: (Throwable, FiniteDuration) => F[Unit]): OnError[F, Throwable] = {
+    (error: Throwable, status: Retry.Status, decision: Decision) => {
       decision match {
         case OnError.Decision.Retry(delay) =>
-          log.warn(s"failed, retrying in $delay, error: $error", error)
```

Comment:
This actually already does what we want, so we don't need to add anything new to master branch. What we need is to back-port the current implementation to CE2 implementation. There's no branch, unfortunately, so you'll first have to make a new series/2.x branch, from [here](https://github.com/evolution-gaming/retry/commit/d1cc39654d343e8cc575be30ce55530718565b69) I assume.